### PR TITLE
Feature/clean up and travis

### PR DESCRIPTION
**Pull request by rolfvdhulst**
Clean up old files and enable Travis tests again.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# This travis image builds roboteam_suite and runs the target 'world_tests'.
```
Comment (mrlukasbos) [inline .travis.yml]: this documentation is not completely correct

File: CMakeLists.txt [hunk 2]
```diff
@@ -7,52 +7,86 @@ include(cotire)
...
--g \
-"
-        )
-
-
-
 
 add_library(roboteam_utils STATIC
         src/Polygon.cpp
+        include/roboteam_utils/Polygon.h
```
Comment (mrlukasbos) [inline CMakeLists.txt]: is it necessary to put the .h files here as well? IIRC this should not be needed

File: CMakeLists.txt [hunk 3]
```diff
@@ -7,52 +7,86 @@ include(cotire)
...
+        test/normaliseTest.cpp
+        test/OptimizationTest.cpp
+        test/PolygonTest.cpp
+        test/PositionTest.cpp
+        test/Vector2Test.cpp
+        )
+
+if (UNIX AND NOT APPLE)
+    target_link_libraries(utils_tests
+            PRIVATE roboteam_utils
```
Comment (mrlukasbos) [inline CMakeLists.txt]: indentation is off

File: include/roboteam_utils/Arc.h [hunk 4]
```diff
@@ -13,7 +13,7 @@ namespace rtt {
  * This is because, appearantly, the fmodl function only has float precision even though it should support
  * 128-bit long doubles...
  */
-    constexpr double ARC_MAX = M_PI - std::numeric_limits<float>::epsilon();
+    constexpr double ARC_MAX = 2*M_PI - std::numeric_limits<float>::epsilon();
```
Comment (mrlukasbos) [inline include/roboteam_utils/Arc.h]: what does this change mean?

File: include/roboteam_utils/BiMap.h [hunk 5]
```diff
@@ -69,13 +69,13 @@ namespace rtt {
          * @return false If key is not found
          */
```
Comment (mrlukasbos) [inline include/roboteam_utils/BiMap.h]: I suppose these changes are an attempt to make this class compile. I saw you made an issue to do it later.

File: include/roboteam_utils/Optimization.h [hunk 6]
```diff
@@ -69,7 +69,7 @@ namespace rtt {
          * the same upper and lower limits.
          */
         GradientDescent(ScoreFunction scorer, Data initialData, Num initialStep, Num minimum, Num maximum)
-                : scorer(scorer), data(data), lastScore(-99999), justFlipped(true), iterationCount(0) {
+                : scorer(scorer), data(initialData), lastScore(-99999), justFlipped(true), iterationCount(0) {
```
Comment (mrlukasbos) [inline include/roboteam_utils/Optimization.h]: what does this change mean?

File: test/Vector2Test.cpp [hunk 7]
```diff
@@ -68,7 +67,7 @@ TEST(VectorTests, math) {
     ASSERT_DOUBLE_EQ(def.dot(tenten), 0);
     ASSERT_DOUBLE_EQ(tenten.dot(fivezero), 50);
     
-    ASSERT_DOUBLE_EQ(M_PI_4l, tenten.angle());
+    //ASSERT_DOUBLE_EQ(M_PI_4, tenten.angle()); //TODO: somehow is not compiling on macOS
```
Comment (mrlukasbos) [inline test/Vector2Test.cpp]: could be that the M_PI_4 macro is not present for clang

File: include/roboteam_utils/BiMap.h (hunk 5 above)
Comment (rolfvdhulst) [reply include/roboteam_utils/BiMap.h]: Yes.They don't give compilation errors now but when you call the constructors somewhere they give linker errors. Still not ideal.

File: include/roboteam_utils/Optimization.h (hunk 6 above)
Comment (rolfvdhulst) [reply include/roboteam_utils/Optimization.h]: Previously it was self-assigning the field (data was initialized with data), this is a bug. This seems like the most sensible fix.

File: test/Vector2Test.cpp (hunk 7 above)
Comment (rolfvdhulst) [reply test/Vector2Test.cpp]: Something like that. I do not know exactly why it's breaking as my compiler just finds it locally.

File: include/roboteam_utils/Arc.h (hunk 4 above)
Comment (rolfvdhulst) [reply include/roboteam_utils/Arc.h]: This is one I'm not entierly sure about. Tests were failing, but also the range of this class in the calculations was assumed to be 0 to 2 PI, so I think this is the correct solution.

File: CMakeLists.txt (hunk 2 above)
Comment (rolfvdhulst) [reply CMakeLists.txt]: It's not necessary but consistent with the rest of our codebase. It also helps if we ever want to port (it does actually make a small difference).

File: CMakeLists.txt (hunk 3 above)
Comment (rolfvdhulst) [reply CMakeLists.txt]: Hmm? This is correct indentation for CMAKE?

File: include/roboteam_utils/BiMap.h (hunk 5 above)
Comment (rolfvdhulst) [reply include/roboteam_utils/BiMap.h]: They aren't used anywhere though, so it's not really important for now.

File: include/roboteam_utils/Arc.h (hunk 4 above)
Comment (rolfvdhulst) [reply include/roboteam_utils/Arc.h]: I looked into it a bit further and yeah... this class needs a lot more help. I'm pretty sure this is correct as in normalize() the angle is clamped between 0 and 2M_PI and not between 0 and PI.

File: include/roboteam_utils/BiMap.h [hunk 8]
```diff
@@ -69,13 +69,13 @@ namespace rtt {
...
         bool removeByLeft(const T &key) {
-            auto it = leftSet.find({left, nullptr});
+            auto it = leftSet.find({key, nullptr});
```
Comment (Yuhanun) [inline include/roboteam_utils/BiMap.h]: This entire structure is basically lookup in sets, meaning it's an std::map

File: test/Vector2Test.cpp (hunk 7 above)
Comment (Yuhanun) [reply test/Vector2Test.cpp]: Any of the pi-constants are non-standard :)

File: include/roboteam_utils/BiMap.h (hunk 8 above)
Comment (rolfvdhulst) [reply include/roboteam_utils/BiMap.h]: yeah.. It's unused so if it's up to me we just remove the class (as it also actually throws a compile error when you try to instantiate the template)

**Review by RobotJente — APPROVED** [review]
Looks good to me!

**Comment by RobotJesse** [comment]
Is this ready for merge?

**Comment by rolfvdhulst** [comment]
> 
> 
> Is this ready for merge?

I am not entirely sure. It might break the Arc class but we would need to test that. For the rest there's nothing dangerous in this PR.

**Comment by rolfvdhulst** [comment]
I don't have time to do it now but I suggest we revert the changes to the arc class and comment out the arc test for now. Someone needs to look at that class in more detail and fix problems.

File: include/roboteam_utils/BiMap.h (hunk 8 above)
Comment (emielsteerneman) [reply include/roboteam_utils/BiMap.h]: Lets remove it. We can always reimplement it if needed

File: include/roboteam_utils/Arc.h (hunk 4 above)
Comment (rolfvdhulst) [reply include/roboteam_utils/Arc.h]: Ok that's good. Still, someone needs to fix this class as there are unfinished functions etc.

**Review by emielsteerneman — APPROVED** [review]
Looks ready to me. BiMap can be removed but that's not needed

**Pull request merged**